File: nodejs/src/decorator/WhipCreamDecorator.ts
import BeverageDecorator from "./BeverageDecorator";

class WhipCreamDecorator extends BeverageDecorator {
    getDescription(): string {
        return `${this.beverage.getDescription()} + Kem tươi`;
    }
    getCost(): number {
        return this.beverage.getCost() + 15000;
    }
}

export default WhipCreamDecorator;

File: spring/demo/src/main/java/com/example/demo/decorator/WhipCreamDecorator.java
package com.example.demo.decorator;

public class WhipCreamDecorator extends BeverageDecorator {
    public WhipCreamDecorator(Beverage beverage) {
        super(beverage);
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + " + Kem tươi";
    }

    @Override
    public int getCost() {
        return beverage.getCost() + 15000;
    }
}